src/sentry/runner/initializer.py

```diff
@@ -327,6 +327,9 @@ def initialize_app(config: dict[str, Any], skip_service_validation: bool = False
 
     configure_structlog()
 
+    if settings.ENVIRONMENT == "production":
+        logging.raiseExceptions = False
```

Comment:
perhaps simpler is `logging.raiseExceptions = settings.DEBUG`

Reply:
@asottile-sentry is `settings.DEBUG` true in _all_ non-prod envs, or only in local-dev?  We'd want it to still blow up in CI and such when something is wrong with logging. 

https://github.com/getsentry/sentry/blob/0d84be95c9f41fb84075b6b97dce3230c6bf3ace/src/sentry/conf/server.py#L100-L104

Reply:
yeah it's only development or production I believe

Reply:
do we want to blow it up in our nearly non-existing staging environment?

Reply:
I deleted staging like 6 months ago

Reply:
I'm waffling on this a bit. IMO it still makes sense to have `IS_PROD` even is _now_ `IS_PROD = not IS_DEV`, because it shows clear intent for what we mean. Also would make our life easier in the future, should we decide to differentiate non-prod envs more.

Reply:
OK, I don't feel strongly about this, so I got rid of it, and changed to using only `settings.DEBUG`